File: LeetCode/735/main.py
class Solution:
    def asteroidCollision(self, arr: List[int]) -> List[int]:
        
        n = len(arr)
        ans = []

        i = 0

        while(i < n) :
            
            if(len(ans) == 0) :
                ans.append(arr[i])
            else :
                prev, cur = ans[-1], arr[i]

                if(prev < 0 or cur > 0) :
                    ans.append(cur)
                else :
                    if(prev == abs(cur)) :
                        ans.pop()
                    elif(prev < abs(cur)):
                        ans.pop()
                        i -= 1
                
            i += 1

        return ans

File: LeetCode/735/main.cpp
class Solution {
public:
  vector<int> asteroidCollision(vector<int> &arr) {
    vector<int> ans;
    int i = 0;
    int n = arr.size();

    while (i < n) {
      if (ans.size() == 0) {
        ans.emplace_back(arr[i]);
      } else {
        int prev = ans[ans.size() - 1];
        int cur = arr[i];

        if (prev < 0 || cur > 0) {
          ans.emplace_back(cur);
        } else {
          if (prev == abs(cur)) {
            ans.pop_back();
          } else if (prev < abs(cur)) {
            ans.pop_back();
            i--;
          }
        }
      }
      i++;
    }

    return ans;
  }
};